### DISCUSSION TITLE
useGeolocation not working in component setup script

### DISCUSSION THREAD

**Discussion by ethanhanderson** [vueuse/vueuse#2128] [Q&A]
I am trying to use the `useGeolocation` method and it works fine in the component template `<span>{{coords}}</span`, but when I try to use it in the component setup script, the properties of `coords` are as follows:
```json
{
    "accuracy": 0,
    "latitude": Infinity,
    "longitude": Infinity,
    "altitude": null,
    "altitudeAccuracy": null,
    "heading": null,
    "speed": null
}
```

I have tried wrapping the method in the `reactive()` function, but that doesn't work either.

Any help would be greatly appreciated!

**Comment by khawarizmus** [comment]
I faced the same problem today and managed to solve it. you can see [here](https://discord.com/channels/937808017016119440/937969993964978197/1051397909574324335)

you need to use the controls that come with `useGeolocation` like this:

```ts
const { coords, error, resume, pause } = useGeolocation({
  enableHighAccuracy: true,
});

onBeforeUnmount(() => {
  pause();
});

function locateUser() {
  if (!error.value) {
    resume();
    setTimeout(() => {
      console.log("coordinates", coords.value);
      // do some logic to check if we got actual values not null of infinity
      // useMapBoxReverseGeocoder()
    }, 5000);
  } else {
    emitter.emit("error", error.value.message);
  }
}
```
